framework/src/systems/NonlinearSystemBase.C

```diff
@@ -3250,6 +3256,15 @@ NonlinearSystemBase::computeScalingJacobian(NonlinearImplicitSystem & sys)
     applyScalingFactors(inverse_scaling_factors);
     if (auto displaced_problem = _fe_problem.getDisplacedProblem().get())
       displaced_problem->systemBaseNonlinear().applyScalingFactors(inverse_scaling_factors);
+
+    // Now it's essential that we reset the sparsity pattern of the matrix
+    petsc_matrix.reset_preallocation();
+
+#else
+    mooseWarning("Automatic scaling requires a PETSc version of 3.8.0 or greater, so no automatic "
```

Comment:
3.9.4

Reply:
So this warning breaks a few tests on the minimum compiler since it is running 3.7.6 still.